system/index.js

```diff
@@ -30,54 +48,77 @@ module.exports = async (message) => {
   const payload = message.payload;
   // Remap headers to a standard JS object
   const headers = message.headers.toRiffHeaders();
-  Object.keys(headers).map((key) => {headers[key] = message.headers.getValue(key)});
-  if (DEBUG) {
-    console.log('==System Function Start==');
-    console.log(`HEADERS: ${JSON.stringify(headers)}`);
-    console.log(`PAYLOAD: ${JSON.stringify(payload)}`);
-    console.log(`MIDDLEWARE_FUNCTION_URI: ${MIDDLEWARE_FUNCTION_URI}`);
-    console.log('==Middleware Function(s) Start==');
-  }
+  Object.keys(headers).map((key) => { headers[key] = message.headers.getValue(key) });
+  const requestID = headers['x-request-id']
```

Comment:
if we're not using `requestID` other than to pass it to the `initLogging` function, then I wonder if we just inline it.

Reply:
> 1. requestId header:  the value we used was "X-Request-Id", note the capitalized letter at beginning of each word, maybe thhee header name is case insensitive, but I think it's better to be consistent

Thanks for the heads!  We'll need to change this @murphybytes, because key 


> 2. Is the logger being created inside node-function buildpack?  which runs AFTER the middleware buildpack, does that mean from middleware, we will not have access to the logger?  I guess I was under the impression that the logger init was done as part of the engine rather than a buildpack?

@jqian2017 the Logger will be created when the function is invoked inside the system function.  Here's the order:
1. Function is invoked
2. The invoker executes the `System` function
3. The `System` function creates the logger
4) The `System` function executes the middleware
5) The `System` function takes middleware result and adds the logger to the end of the args list

If the middleware needs access to the logger we could potentially pass it along through the the middleware chain.  Is this something you'd like to have?

6)

Reply:
I think the purpose is to have all the logs contain the same requestId, so trace can be stitched together... so I'd like to have access to the logger object from middleware in case I need to log something there.

FYI, @cwallsfdc

Reply:
@elbandito 
Is the logger is typed and defined in salesforce-sdk? (as both middleware and customer uses typescript)

Reply:
@jqian2017 with regards to the mixed case X-Request-Id, Riff normalizes the header name [here](https://github.com/projectriff/node-message/blob/master/lib/headers.js#L131) so when I get it, it's lower case.